bitmind/validator/forward.py

```diff
@@ -60,15 +78,11 @@ async def forward(self):
         bt.logging.info('sampling real image')
 
         label = 0
-        real_dataset_index, source_dataset = sample_dataset_index_name(self.real_image_datasets)
-        real_dataset = self.real_image_datasets[real_dataset_index]
-        samples, idx = real_dataset.sample(k=1)  # {'image': PIL Image ,'id': int}
-        sample = samples[0]
-
-        wandb_data['dataset'] = source_dataset
-        wandb_data['image_index'] = idx[0]
-        wandb_data['image_name'] = sample['id']
-
+        random_idx = np.random.randint(0, self.total_real_images)
+        source_dataset, local_index = sample_random_real_image(self.real_image_datasets, self.total_real_images)
+        wandb_data['source_dataset'] = source_dataset.huggingface_dataset_name
+        wandb_data['source_image_index'] = local_index
+        sample = source_dataset[local_index]
```

Comment:
Q: Why are we not retrying for real images v.s. retrying for fake images?

Reply:
retries are built into sample_random_real_image, but for synthetic images there's an additional retry mechanism for failed generation